### PR TITLE
Add minimum RSSI check to ble presence

### PR DESCRIPTION
**Pull request by nielsnl68**
# What does this implement/fix?

I do have multiple BLE_PRESENCE sensors around the house and i liked to set when the sensor should say my etag was marked as present.
This PR will implement this behavier.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#2800

## Test Environment

- [ ] ESP32
- [x] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

binary_sensor:
  # Presence based on MAC address
  - platform: ble_presence
    mac_address: AC:37:43:77:5F:4C
    name: "ESP32 BLE Tracker"
    min_rssi: -80
```

## Checklist:
  - [ ] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/ble_presence/binary_sensor.py [hunk 1]
```diff
@@ -51,6 +53,9 @@ async def to_code(config):
     await cg.register_component(var, config)
     await esp32_ble_tracker.register_ble_device(var, config)
 
+    if CONF_MIN_RSSI in config:
+        cg.add(var.set_minimum_rssi(config[CONF_MIN_RSSI].as_hex))
```
Comment (jesserockz) [inline esphome/components/ble_presence/binary_sensor.py]: `as_hex` here is a little bit useless
```suggestion
        cg.add(var.set_minimum_rssi(config[CONF_MIN_RSSI]))
```

File: esphome/components/ble_presence/binary_sensor.py [hunk 2]
```diff
@@ -37,6 +38,7 @@ def _validate(config):
             cv.Optional(CONF_IBEACON_MAJOR): cv.uint16_t,
             cv.Optional(CONF_IBEACON_MINOR): cv.uint16_t,
             cv.Optional(CONF_IBEACON_UUID): cv.uuid,
+            cv.Optional(CONF_MIN_RSSI): cv.uint16_t,
```
Comment (jesserockz) [inline esphome/components/ble_presence/binary_sensor.py]: The rssi reported is negative and should be using this:
```suggestion
            cv.Optional(CONF_MIN_RSSI): cv.decibel,
```

File: esphome/components/ble_presence/ble_presence_device.h [hunk 3]
```diff
@@ -41,12 +41,19 @@ class BLEPresenceDevice : public binary_sensor::BinarySensorInitiallyOff,
     this->check_ibeacon_minor_ = true;
     this->ibeacon_minor_ = minor;
   }
+  void set_minimum_rssi(uint16_t rssi) {
```
Comment (jesserockz) [inline esphome/components/ble_presence/ble_presence_device.h]:
```suggestion
  void set_minimum_rssi(int rssi) {
```

File: esphome/components/ble_presence/ble_presence_device.h [hunk 4]
```diff
@@ -96,17 +103,21 @@ class BLEPresenceDevice : public binary_sensor::BinarySensorInitiallyOff,
...
 
   esp32_ble_tracker::ESPBTUUID ibeacon_uuid_;
-  uint16_t ibeacon_major_;
-  bool check_ibeacon_major_;
-  uint16_t ibeacon_minor_;
-  bool check_ibeacon_minor_;
+  uint16_t ibeacon_major_{0};
+  uint16_t ibeacon_minor_{0};
+
+  uint16_t minimum_rssi_{0};
```
Comment (jesserockz) [inline esphome/components/ble_presence/ble_presence_device.h]:
```suggestion
  int minimum_rssi_{0};
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

**Review by jesserockz — APPROVED** [review]

**Comment by nielsnl68** [comment]
I need some time to test this better :(

**Review by jesserockz — APPROVED** [review]

**Pull request merged**